src/bindings/python/src/openvino/runtime/ie_api.py

```diff
@@ -268,6 +278,8 @@ def __call__(
         inputs: Any = None,
         share_inputs: bool = True,
         share_outputs: bool = False,
+        *,
+        decode_strings: bool = True,
```

Comment:
I'm a bit lost in our intentions to align `infer` and `__call__` methods. From one hand, I see that `share_outputs` is set to `False` in both `infer` and `__call__` methods which provides alignment on how we can process outputs. But here we are introducing quite a visible difference in generated outputs: decoded or not decoded strings. So, to maximize performance user will still need to set `share_outputs` to `True` for other inputs (not string), so he/she should be aware of these knobs that affect performance. It is not obvious for me what should we prefer in this case: alignment in APIs or performance in one of the places.

Reply:
@apaniukov can you share more detailed justification of the request?

@slyalin I see the point. How about the solution that will decode in both scenarios, with optional disabling of functionality?
Let both `CompiledModel` and `InferRequest` set `decode_strings=True` by default, however anyone that wants to optimize the pipeline will be able to switch it off. @apaniukov is this design OK for your use-cases? It's always less confusion on users side but gives the requested flexibility.

Reply:
@jiwaszki Yes, this fits my use case too. We can emphasize the existence of that option in our samples.